lib/remoteproc/remoteproc.c

```diff
@@ -36,11 +36,16 @@ remoteproc_get_mem(struct remoteproc *rproc, const char *name,
 {
 	struct metal_list *node;
 	struct remoteproc_mem *mem;
+	unsigned int str_size;
 
 	metal_list_for_each(&rproc->mems, node) {
 		mem = metal_container_of(node, struct remoteproc_mem, node);
 		if (name) {
-			if (!strncmp(name, mem->name, sizeof(mem->name)))
+			if (strlen(name) < sizeof(mem->name))
```

Comment:
@arnop2 strncmp() will stop the comparison as soon as it hits the null terminator in either string, or when the specified number of characters have been compared. Therefore it is safe to use in this situation. 
There is another bug here, namely that mem->name can be non-null terminated.

Reply:
@edmooring  Could you tell me if your comment is related to the the last version of the code using strcmp? In this case, please,  could you comment in code instead of replying of the conversation which is outdated? easier to follow :)
If name is not terminated by a null character and  sizeof (mem->name) > sizeof(name) you can access to the memory following the one reserved for name buffer.  This is the point highlighted by MISRA. Something strange (for my unsderstanding) is that MISRA accepts the use of strcmp instead...